roles/openshift-route-status/tasks/main.yml

```diff
@@ -1,31 +1,26 @@
 ---
 
-- name: "Lookup route url for {{ route }}"
-  shell: "oc get route {{ route }} -o jsonpath='{ .spec.host }'"
-  register: url
-  when:
-  - route is defined
-  - route|trim != ''
+- name: Verify Required Parameters
+  fail:
+    msg: "'route', 'protocol' and 'status' Parameters Must be Provided!"
+  when: route is not defined or route|trim == '' or protocol is not defined or protocol|trim == '' or status is not defined or status|trim == ''
 
-- name: "Set default delay time"
+- name: "Set the target namespace option if supplied"
   set_fact:
-    delay: "{{ delay | default(5) }}"
+    target_namespace: "-n {{ namespace }}"
+  when:
+  - namespace is defined
+  - namespace|trim != ''
 
-- name: "Set default number of retries"
-  set_fact:
-    retries: "{{ retries | default(3) }}"
+- name: "Lookup route url for {{ route }}"
+  shell: "oc get route {{ route }} {{ target_namespace }} -o jsonpath='{ .spec.host }'"
+  register: url
 
 - name: "Wait for {{ protocol }}://{{ url.stdout }} to respond with status: {{ status }}"
   uri:
     url: "{{ protocol }}://{{ url.stdout }}"
+    headers: "{{ headers | default({}) }}"
```

Comment:
As you are moving much of the other parameters to the defaults/main.yml file, can you do so with `headers` as well?